app/src/main/java/com/github/se/icebreakrr/model/notification/EngagementNotificationManager.kt

```diff
@@ -78,19 +90,51 @@ class EngagementNotificationManager(
    * and processes them if the user is discoverable.
    */
   private fun checkNearbyUsersForCommonTags() {
-    profilesViewModel.getSelfProfile {
-      val selfProfile = profilesViewModel.selfProfile.value ?: return@getSelfProfile
-      val selfLocation = selfProfile.location ?: return@getSelfProfile
+    Log.e(TAG, "Checking for nearby users")
```

Comment:
be aware that you have a lot of logs in this file that do not indicate errors and that are logged very often, and might remove them for the final implementation